File: Tyche.Backend/Pipel.Tyche.Pool/Pipel.Tyche.Pool/Domain/PoolGambler.fs
namespace Pipel.Tyche.Pool.Domain

open Pipel.Tyche.Pool.Domain
open Pipel.Type

type PoolGamblerPK =
    { PoolPK: PoolPK
      GamblerPK: GamblerPK }

type PoolGambler =
    { PoolGamblerPK: PoolGamblerPK
      GamblerEmail: Email
      Score: PositiveInt option
      CurrentPosition: PositiveInt option
      BeforePosition: PositiveInt option }

File: Tyche.Android/app/src/main/java/com/pipel/tyche/poolGambler/domain/PoolGambler.kt
package com.pipel.tyche.poolGambler.domain

import com.pipel.core.type.Email
import com.pipel.core.type.PositiveInt
import com.pipel.core.type.Ulid

data class PoolGambler(
    val poolId: Ulid,
    val gamblerId: Ulid,
    val gamblerEmail: Email,
    val score: PositiveInt?,
    val currentPosition: PositiveInt?,
    val beforePosition: PositiveInt?
)